docs/boot-process.md

```diff
@@ -0,0 +1,382 @@
+# Bottlerocket Boot Process
+
+This document describes how Bottlerocket boots, focusing on the systemd target progression and service dependencies.
+
+**Keywords:** boot, systemd, targets, preconfigured, configured, multi-user, fipscheck, drivers, sysinit, services, dependencies, ordering, API system, bootstrap containers, settings, startup, initialization, kernel modules
+
+## Overview
+
+Bottlerocket's boot sequence progresses through six main stages, each represented by a systemd target:
+
+```
+sysinit.target
```

Comment:
I would probably add `local-fs.target` as one of the most important prerequisites to `sysinit.target`.

Quite a lot of Bottlerocket-specific work happens for local storage setup 😀 while the `sysinit` phase is rather vanilla.